backend/src/test/java/com/nikogrid/backend/repositories/UserRepositoryTest.java

```diff
@@ -0,0 +1,251 @@
+
+        List<User> allUsers = userRepository.findAll();
+
+        assertThat(allUsers).hasSize(2);
+        assertThat(allUsers).extracting(User::getEmail)
+                .containsExactlyInAnyOrder("test1@example.com", "test2@example.com");
+    }
+
+    @Test
+    void testDeleteUser() {
+        User savedUser = userRepository.save(testUser1);
+        UUID userId = savedUser.getId();
+        entityManager.flush();
+
+        userRepository.deleteById(userId);
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findById(userId);
+        assertThat(foundUser).isNotPresent();
+    }
+
+    @Test
+    void testUpdateUser() {
+        User savedUser = userRepository.save(testUser1);
+        entityManager.flush();
+
+        savedUser.setEmail("updated@example.com");
+        savedUser.setPassword("newPassword");
+
+        User updatedUser = userRepository.save(savedUser);
+        entityManager.flush();
+
+        assertThat(updatedUser.getEmail()).isEqualTo("updated@example.com");
+        assertThat(updatedUser.getPassword()).isEqualTo("newPassword");
+
+        Optional<User> foundUser = userRepository.findByEmail("updated@example.com");
+        assertThat(foundUser).isPresent();
+        assertThat(foundUser.get().getPassword()).isEqualTo("newPassword");
+    }
+
+    @Test
+    void testUserWithMultipleRoles() {
+        Role collaboratorRole = Role.builder()
+                .name(RoleName.COLLABORATOR)
+                .build();
+        entityManager.persistAndFlush(collaboratorRole);
+
+        User userWithMultipleRoles = User.builder()
+                .email("multirole@example.com")
+                .password("password")
+                .roles(new ArrayList<>(List.of(adminRole, ownerRole, collaboratorRole)))
+                .created_at(LocalDateTime.now())
+                .build();
+
+        userRepository.save(userWithMultipleRoles);
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findByEmail("multirole@example.com");
+
+        assertThat(foundUser).isPresent();
+        assertThat(foundUser.get().getRoles()).hasSize(3);
+        assertThat(foundUser.get().getRoles()).extracting(Role::getName)
+                .containsExactlyInAnyOrder(RoleName.ADMIN, RoleName.OWNER, RoleName.COLLABORATOR);
+    }
+
+    @Test
+    void testUniqueEmailConstraint() {
+        userRepository.save(testUser1);
+        entityManager.flush();
+
+        User duplicateEmailUser = User.builder()
+                .email("test1@example.com") // Same email as testUser1
+                .password("differentPassword")
+                .roles(new ArrayList<>(List.of(ownerRole)))
+                .created_at(LocalDateTime.now())
+                .build();
+
+        assertThatThrownBy(() -> {
+            userRepository.save(duplicateEmailUser);
+            entityManager.flush();
+        }).isInstanceOf(Exception.class); // Could be DataIntegrityViolationException or similar
```

Comment:
Catching an `Exception` isn't really a test as anything throwing would cause the test to pass. You should narrow down the exception.